code/game/machinery/vendors/vending.dm

```diff
@@ -1037,6 +1037,8 @@
 
 	. = FALSE
 
+	if(from_anywhere)
```

Comment:
Something like this is coming in the vendor/general tilt refactor as well, watch that space